pennylane/compiler/python_compiler/transforms/quantum/parity_synth.py

```diff
@@ -0,0 +1,353 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains the implementation of the ``ParitySynth`` compiler pass. Given that it
+operates on the phase polynomial representation of subcircuits, the implementation splits into
+an xDSL-agnostic synthesis functionality and an integration thereof into xDSL."""
+
+from dataclasses import dataclass
+from itertools import product
+
+import networkx as nx
+import numpy as np
+from xdsl import context, passes, pattern_rewriter
+from xdsl.dialects import arith, builtin, func
+from xdsl.ir import SSAValue
+from xdsl.rewriter import InsertPoint
+
+from .....transforms.intermediate_reps.rowcol import _rowcol_parity_matrix
```

Comment:
Oof

I think for imports from outside `qml.compiler.python_compiler` absolute imports might make more sense
```suggestion
from pennylane.transforms.intermediate_reps.rowcol import _rowcol_parity_matrix

```